extensions/scala/deployment/src/main/java/io/quarkus/scala/deployment/ScalaCompilationProvider.java

```diff
@@ -27,9 +27,8 @@ public Set<String> handledExtensions() {
     @Override
     public void compile(Set<File> files, Context context) {
         Settings settings = new Settings();
-        context.getClasspath().stream()
-                .map(File::getAbsolutePath)
-                .forEach(f -> settings.classpath().append(f));
+        context.getClasspath().forEach(file -> settings.classpath().append(file.getAbsolutePath()));
+        context.getReloadableClasspath().forEach(file -> settings.classpath().append(file.getAbsolutePath()));
```

Comment:
Scala compiler settings does not expose the file manager implementation as Java, the solution is to add the reloadable paths to the classpath setting. 

TODO: JavaConverters is [deprecated](https://github.com/scala/scala/blob/v2.13.8/src/library/scala/collection/JavaConverters.scala#L79), the recommendation is to use `scala.jdk.CollectionConverters`. I ask for your opinion, as it may or may not be necessary to change.
```java
//run.compile(JavaConverters.asScalaSet(fileSet).toList());
run.compile(CollectionConverters.SetHasAsScala(fileSet).asScala().toList());
```